src/main/java/net/mcreator/ui/views/editor/image/clipboard/ClipboardManager.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2023, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.ui.views.editor.image.clipboard;
+
+import net.mcreator.ui.views.editor.image.ImageMakerView;
+import net.mcreator.ui.views.editor.image.clipboard.transferables.CanvasTransferable;
+import net.mcreator.ui.views.editor.image.clipboard.transferables.LayerTransferable;
+import net.mcreator.ui.views.editor.image.layer.Layer;
+import net.mcreator.ui.views.editor.image.canvas.Canvas;
+import net.mcreator.ui.views.editor.image.versioning.change.Modification;
+
+import java.awt.*;
+import java.awt.datatransfer.*;
+import java.io.IOException;
+import java.util.UUID;
+
+public class ClipboardManager implements ClipboardOwner {
+	private final Clipboard clipboard;
+	private final ImageMakerView imageMakerView;
+
+	public ClipboardManager(ImageMakerView imageMakerView) {
+		this.imageMakerView = imageMakerView;
+		clipboard = Toolkit.getDefaultToolkit().getSystemClipboard();
+	}
+
+	public void copy() {
+		addLayerToClipboard();
+	}
+
+	public void copyAll() {
+		addCanvasToClipboard();
+	}
+
+	public void cut() {
+		addLayerToClipboard();
+		Layer selected = imageMakerView.getLayerPanel().selected();
+		selected.clearSelection();
+		imageMakerView.getVersionManager().addRevision(new Modification(imageMakerView.getCanvas(), selected));
+	}
+
+	private void addCanvasToClipboard() {
+		addLayerToClipboard(new CanvasTransferable(imageMakerView.getCanvas()));
+	}
+
+	private void addLayerToClipboard() {
+		Layer selected = imageMakerView.getLayerPanel().selected();
+		if (selected != null)
+			addLayerToClipboard(new LayerTransferable(selected));
+	}
+
+	private void addLayerToClipboard(Transferable transferable) {
+		// Prevent crashing on systems that lock clipboards to specific programs.

```

Comment:
https://github.com/MCreator/MCreator/pull/4331 fixes the clipboard bug. @MysteriousWolf could you remove the notice about this in the code now and we will first merge the JDK update?

Reply:
Should be removed now